client/src/components/Chart/index.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import _ from 'lodash';
+import { VictoryChart, VictoryLine, VictoryZoomContainer } from 'victory';
+
+const Chart = (props) => {
+  const [zoomedXDomain, setZoomedXDomain] = useState(props.entireDomain.x);
+
+  const onDomainChange = (domain) => {
+    setZoomedXDomain(domain.x);
+  };
+
+  const getData = () => {
+    const { data, maxPoints } = props;
+    const filtered = data.filter(
+      (d) => d.x >= zoomedXDomain[0] && d.x <= zoomedXDomain[1],
+    );
+    if (filtered.length > maxPoints) {
+      const k = Math.ceil(filtered.length / maxPoints);
+      return filtered.filter((d, i) => i % k === 0);
+    }
+    return filtered;
+  };
+
+  const getZoomFactor = () => {
+    const factor = 10 / (zoomedXDomain[1] - zoomedXDomain[0]);
+    return _.round(factor, factor < 3 ? 1 : 0);
```

Comment:
can you add a comment explaining the numbers? it's not immediately intuitive why the numbers are chosen